devices/Atgm336h/Atgm336h.cs

```diff
@@ -0,0 +1,233 @@
+            _serialDevice.DataBits = 8;
+            _serialDevice.ReadBufferSize = 2048;
+            _serialDevice.ReadTimeout = 2000;
+            _serialDevice.WatchChar = '\r';
+            _serialDevice.DataReceived += SerialDevice_DataReceived;
+            _shouldDispose = true;
+        }
+        
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Atgm336h" /> class.
+        /// </summary>
+        /// <param name="serialPort">Serial port instance.</param>
+        public Atgm336h(SerialPort serialPort)
+        {
+            _serialDevice = serialPort;
+            _serialDevice.DataReceived += SerialDevice_DataReceived;
+            _shouldDispose = false;
+        }
+
+        /// <summary>
+        /// Starts the GPS module by opening the serial device connection.
+        /// </summary>
+        public void Start()
+        {
+            _serialDevice.Open();
+        }
+
+        /// <summary>
+        /// Stops the communication with the GPS module by closing the serial port.
+        /// </summary>
+        public void Stop()
+        {
+            _serialDevice.Close();
+        }
+        
+        private static double ConvertToGeoLocation(string data, string direction)
+        {
+            var degreesLength = data.Length > 10 ? 3 : 2;
+
+            var degrees = double.Parse(data.Substring(0, degreesLength));
+            var minutes = double.Parse(data.Substring(degreesLength));
+
+            var result = degrees + (minutes / 60);
+
+            if (direction == "S" || direction == "W")
+            {
+                return -result;
+            }
+
+            return result;
+        }
+
+        private static Mode ConvertToMode(string data)
+        {
+            switch (data)
+            {
+                case "M":
+                    return Mode.Manual;
+                case "A":
+                    return Mode.Auto;
+            }
+
+            throw new Exception();
+        }
+
+        private static Fix ConvertToFix(string data)
+        {
+            switch (data)
+            {
+                case "1":
+                    return Fix.NoFix;
+                case "2":
+                    return Fix.Fix2D;
+                case "3":
+                    return Fix.Fix3D;
+            }
+
+            throw new Exception();
+        }
+        
+        private void SerialDevice_DataReceived(object sender, SerialDataReceivedEventArgs e)
+        {
+            var serialDevice = (SerialPort)sender;
+            if (serialDevice.BytesToRead == 0)
+            {
+                return;
+            }
+
+            var buffer = new byte[serialDevice.BytesToRead];
+            var bytesRead = serialDevice.Read(buffer, 0, buffer.Length);
+            var stringBuffer = Encoding.UTF8.GetString(buffer, 0, bytesRead);
+            var commands = stringBuffer.Split(serialDevice.WatchChar);
+            foreach (var command in commands)
+            {
```

Comment:
as you are manipulating strigs, that then may not be complete, I would go for a try/catch for this function.

Reply:
done + added exception event